src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -136,7 +140,50 @@ public FoldingRangeRegistrationOptions GetRegistrationOptions(FoldingRangeCapabi
                 mappedRanges.AddRange(ranges);
             }
 
-            return new Container<FoldingRange>(mappedRanges);
+            var finalRanges = FinalizeFoldingRanges(mappedRanges, codeDocument);
+            return new Container<FoldingRange>(finalRanges);
+        }
+
+        private IEnumerable<FoldingRange> FinalizeFoldingRanges(List<FoldingRange> mappedRanges, RazorCodeDocument codeDocument)
+        {
+            // Don't allow ranges to be reported if they aren't spanning at least one line
+            var validRanges = mappedRanges.Where(r => r.StartLine < r.EndLine);
```

Comment:
I feel like the fact that we got "invalid" ranges is worthy of a `_logger.Warning()`, or at least a `_logger.Information()`. Surely our partners should not give us a folding range which cannot fold?

Reply:
I'll do a follow up here. Even if we are getting invalid ranges we want to be resilient. Logging could get very verbose very quickly for large files here.